pkg/util/struct.go

```diff
@@ -0,0 +1,48 @@
+package util
+
+import "reflect"
+
+// VisitAllStringFields visits all string fields in the given pointer to struct
+// and allow modifying them with visitor function
+func VisitAllStringFields(obj interface{}, visitor func(string) string) {
```

Comment:
why not make this take a type argument so it can generically visit any field type?

and why not have the visitor take the val object and do what it wants with it, rather than having the visitor return a value to be set?   I don't think this is a true visitor pattern.